### ISSUE TITLE
[Bug] LoginsStorageException: Using DatabaseLoginsStorage without unlocking first

### ISSUE THREAD

**Issue by AndiAJ** [mozilla-mobile/fenix#8854]
## Steps to reproduce
1. Navigate to Vimeo.com
2. Perform a search

## Alternative STR
1. Navigate to https://www.google.com/search?q=vimeo%20f1&ie=utf-8&oe=utf-8&client=firefox-b-m
2. Open one of the featured Vimeo videos

### Expected behavior
Vimeo should work properly.

### Actual behavior
Crash

### Device information
* Android device: OnePlus A3 (Android 6.0.1)
* Fenix version:  
 • Fenix 4.0.0-beta.2
• Nightly Build #20590607 from 2/28

### Notes
❗ Sometimes reproducible only after a couple of attempts.

► Video
![20200228-142340](https://user-images.githubusercontent.com/51314259/75549616-c2bc2500-5a38-11ea-9e0d-535aedf42bfe.gif)

► Video
![20200228-143141](https://user-images.githubusercontent.com/51314259/75549696-ee3f0f80-5a38-11ea-8558-3bbd434a1e2a.gif)

► Crashlog
```kotlin
 --------- beginning of crash
02-28 14:25:34.136 7975-8026/? E/AndroidRuntime: FATAL EXCEPTION: DefaultDispatcher-worker-6
    Process: org.mozilla.fenix.beta, PID: 7975
    mozilla.appservices.logins.LoginsStorageException: Using DatabaseLoginsStorage without unlocking first
        at mozilla.appservices.logins.DatabaseLoginsStorage.checkUnlocked(DatabaseLoginsStorage.kt:2)
        at mozilla.appservices.logins.DatabaseLoginsStorage.access$checkUnlocked(DatabaseLoginsStorage.kt:1)
        at mozilla.appservices.logins.DatabaseLoginsStorage.getByBaseDomain(DatabaseLoginsStorage.kt:4)
        at mozilla.components.service.sync.logins.AsyncLoginsStorageAdapter$getByBaseDomain$1.invokeSuspend(AsyncLoginsStorage.kt:4)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:2)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:15)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:1)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:15)
02-28 14:25:34.136 7975-8026/? E/AbstractTracker: Can't create handler inside thread that has not called Looper.prepare()
02-28 14:25:34.136 7975-8026/? D/AppTracker: App Event: crash
02-28 14:25:34.138 7975-8026/? E/AbstractTracker: mTrackerAsyncQueryHandler is null
02-28 14:25:34.150 1298-1317/? V/BoostFramework: BoostFramework() : mPerf = com.qualcomm.qti.Performance@1159bb3
02-28 14:25:34.151 1298-1317/? I/ActivityManager: Start proc 8327:org.mozilla.fenix.beta:mozilla.components.lib.crash.CrashReporter/u0a221 for activity org.mozilla.fenix.beta/mozilla.components.lib.crash.prompt.CrashReporterActivity
02-28 14:25:34.158 1298-1334/? V/BoostFramework: BoostFramework() : mPerf = com.qualcomm.qti.Performance@2a3767a
02-28 14:25:34.158 1298-1334/? V/BoostFramework: BoostFramework() : mPerf = com.qualcomm.qti.Performance@a0ece2b
02-28 14:25:34.164 1298-1334/? W/ActivityManager:  Dismiss app error dialog : org.mozilla.fenix.beta
02-28 14:25:34.164 1298-1334/? W/AppOps: Finishing op nesting under-run: uid 1000 pkg android code 24 time=0 duration=0 nesting=0
02-28 14:25:34.166 7975-8026/? I/Process: Sending signal. PID: 7975 SIG: 9
02-28 14:25:34.175 1298-8326/? D/DropBoxManagerService: file :: /data/system/dropbox/data_app_crash@2020-02-28-14_25_34_142.txt
02-28 14:25:34.243 1298-1317/? D/GraphicsStats: Buffer count: 6
02-28 14:25:34.243 1298-2233/? I/WindowState: WIN DEATH: Window{f38cf34 u0 SurfaceView}
02-28 14:25:34.243 1298-1317/? I/ActivityManager: Process org.mozilla.fenix.beta (pid 7975) has died
02-28 14:25:34.246 8063-8063/? I/ServiceChildProcess: Service has been unbound. Stopping.
02-28 14:25:34.246 1298-1717/? W/InputDispatcher: channel 'c73c600 org.mozilla.fenix.beta/org.mozilla.fenix.beta.App (server)' ~ Consumer closed input channel or an error occurred.  events=0x9
02-28 14:25:34.246 1298-1717/? E/InputDispatcher: channel 'c73c600 org.mozilla.fenix.beta/org.mozilla.fenix.beta.App (server)' ~ Channel is unrecoverably broken and will be disposed!
02-28 14:25:34.249 1298-2661/? I/WindowState: WIN DEATH: Window{c73c600 u0 org.mozilla.fenix.beta/org.mozilla.fenix.beta.App}
02-28 14:25:34.249 8039-8039/? I/ServiceChildProcess: Service has been unbound. Stopping.```

**Comment by st3fan** [comment]
I renamed this bug since it is not Vimeo specific. This is a common crash related to login database locking.

See https://sentry.prod.mozaws.net/operations/fenix-nightly/issues/7380071

**Comment by kbrosnan** [comment]
https://crash-stats.mozilla.com/signature/?java_stack_trace=~mozilla.appservices.logins.LoginsStorageException&product=Fenix&signature=mozilla.appservices.logins.LoginsStorageException%3A%20at%20mozilla.appservices.logins.DatabaseLoginsStorage.checkUnlocked%28DatabaseLoginsStorage.kt%3A2%29

**Comment by pocmo** [comment]
For me it currently reproduces every time on Nightly with STR:
* Browse to vimeo.com
* Click on "Join free" button
* Click on "Log In" tab
* Click into "Email" field
* Refresh the website from the menu

**Comment by ekager** [comment]
AFAIU this needs a fix outside of Fenix (unless we are unlocking/locking the database incorrectly) See: https://github.com/mozilla-mobile/android-components/issues/5715 

If we are seeing enough events here (which it seems like we might be) a release solution is to re-add the logins feature flags for beta/release.

**Comment by ekager** [comment]
@pocmo said this has spiked in the last day or two which makes me think our changes to stop runBlocking in onLoginFetch operations may have either caused or revealed some of these underlying issues. 
We removed runBlocking for a scope.async here https://github.com/mozilla-mobile/android-components/pull/6059/files#diff-e6d788ae9565420bf5f7e833e6779cdeR80

**Comment by ekager** [comment]
Opened https://github.com/mozilla-mobile/fenix/pull/8864 to re-add logins feature flags temporarily. We can uplift this to our beta/release builds

**Comment by grigoryk** [comment]
Should be addressed by https://github.com/mozilla-mobile/fenix/pull/8875

**Comment by abodea** [comment]
I was able to reproduce this issue with other steps on the latest Nightly 3/2 with Samsung Galaxy S10+ (Android 9).
- Launch Fenix 
- Access any website(facebook, twitter, etc)
- Tap on sign-in and add any character there or if there is already something pre-filled delete the data from `email`.
- Fenix crashes.
Reproducible 10/10.

![20200302-143948](https://user-images.githubusercontent.com/42831109/75677394-0363aa80-5c94-11ea-8bfc-4f83d43bac2a.gif)


```kotlin
--------- beginning of crash
2020-03-02 14:32:42.439 4547-4578/? E/AndroidRuntime: FATAL EXCEPTION: DefaultDispatcher-worker-1
    Process: org.mozilla.fenix.nightly, PID: 4547
    mozilla.appservices.logins.LoginsStorageException: Using DatabaseLoginsStorage without unlocking first
        at mozilla.appservices.logins.DatabaseLoginsStorage.checkUnlocked(DatabaseLoginsStorage.kt:2)
        at mozilla.appservices.logins.DatabaseLoginsStorage.access$checkUnlocked(DatabaseLoginsStorage.kt:1)
        at mozilla.appservices.logins.DatabaseLoginsStorage.getByBaseDomain(DatabaseLoginsStorage.kt:4)
        at mozilla.components.service.sync.logins.AsyncLoginsStorageAdapter$getByBaseDomain$1.invokeSuspend(AsyncLoginsStorage.kt:4)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:2)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:15)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:1)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:15)
2020-03-02 14:32:42.439 5725-5725/? D/CS_HdrController: updateMultiWindowState : 0
2020-03-02 14:32:42.439 5725-5725/? D/MdnieScenarioControlService: MultiWindowState : false
2020-03-02 14:32:42.440 9020-9020/? I/SKBD: [ATIM] [ppfockfsk] IMI,25,IR,0
2020-03-02 14:32:42.440 9020-9020/? I/SKBD: cmq [PF_KL][SIIC] getTextBeforeCursor[ 2 ] done  0
2020-03-02 14:32:42.440 5725-6017/? V/WindowManager: Setting visibility of Window{23aaa0c u0 com.sec.android.app.launcher/com.sec.android.app.launcher.activities.LauncherActivity}: true, caller=com.android.server.wm.WindowContainer.sendAppVisibilityToClients:623 com.android.server.wm.AppWindowToken.setClientHidden:460 com.android.server.wm.AppWindowContainerController.setVisibility:440 com.android.server.am.ActivityRecord.setVisibility:1871 com.android.server.am.ActivityRecord.setVisible:1879 
2020-03-02 14:32:42.440 9020-9020/? I/SKBD: [IIManager] IIfo SI [il:0x656e5553] [p:0] [c:0] [t:1] [im:0/0/0/0/0/0] [sC:0] [aC:1] [sS:0] [aS:1] [aP:0] [pW:0] [eM:1] [uR:0] [acI:0] [iT:21] [iO:12000005] [pID:4547] [pO:null]
2020-03-02 14:32:42.440 9020-9020/? I/SKBD: [IIManager] IIfo II [ir:0] [rand:a668714c9cdc4af59b1d42690942896b] [hC:0] [hI:0] [tr:0] [tc:1] [long:0] [cpl:0] [xyD:0/0] [xycD:0/0] [kWHG:0/0/0] [tD:0] [tP:7615] [fil:true] [IC:0/0] [cB:0/0]
2020-03-02 14:32:42.440 5725-6017/? V/WindowOrientationListener: getProposedRotation :0
2020-03-02 14:32:42.440 5725-6017/? V/WindowManager: rotationForOrientationLw(orient=5, last=0); user=0 USER_ROTATION_LOCKED sensorRotation=0 mLidState=-1 mDockMode=0 mHdmiPlugged=false
2020-03-02 14:32:42.440 9020-9020/? I/SKBD: [ATIM] [PF_KL][onCharacterKey]  0
2020-03-02 14:32:42.440 9020-9020/? I/SKBD: arw [PF_KL][onKeyHandleNormalKey::onCharacterKey] 403462
2020-03-02 14:32:42.441 9020-9020/? I/SKBD: ame [PF_KL][onKey] 939807
2020-03-02 14:32:42.443 9020-9020/? I/SKBD: cmq [PF_KL][SIIC] getCursorCapsMode done  0
2020-03-02 14:32:42.443 31813-31813/? W/ActivityThread: handleWindowVisibility: no activity for token android.os.BinderProxy@86631e2
2020-03-02 14:32:42.454 5725-6017/? D/GamePkgDataHelper: notifyAppCreate(), pkgName: org.mozilla.fenix.nightly, sendRet: true
2020-03-02 14:32:42.455 5725-5987/? D/GamePkgDataHelper: getGamePkgData(). org.mozilla.fenix.nightly
2020-03-02 14:32:42.455 5725-5987/? D/GameManagerService: handleMessage(), MSG_APP_CREATE. ignore. pkgName: org.mozilla.fenix.nightly
2020-03-02 14:32:42.455 5725-6017/? D/InputDispatcher: Focused application set to: d61e8a6
2020-03-02 14:32:42.456 5725-6017/? D/GameManagerService: noteResumeComponent(), resumeComponent: org.mozilla.fenix.nightly
2020-03-02 14:32:42.456 5725-5987/? D/GameManagerService: handleForegroundChange(). pkgName: org.mozilla.fenix.nightly, clsName: mozilla.components.lib.crash.prompt.CrashReporterActivity,FgActivityName:org.mozilla.fenix.nightly/mozilla.components.lib.crash.prompt.CrashReporterActivity
2020-03-02 14:32:42.456 5725-5987/? D/GameManagerService:   handleForegroundChange(). same package. game has never resumed yet. ignore
2020-03-02 14:32:42.457 8576-8576/? I/Launcher: onWindowVisibilityChanged : false
2020-03-02 14:32:42.457 5725-6017/? D/MARsPolicyManager: onPackageResumedFG pkgName = org.mozilla.fenix.nightly, userId = 0
```

**Comment by liuche** [comment]
The AC fix for this #8875 should be picked up now, and ready to verify.

**Comment by AndiAJ** [comment]
Hi, **verified as fixed** on both **`Fenix Nightly Build #20640607 from 3/4`** and **`Fenix Beta 4.0.0-beta.3 from 3/3`** using the following devices:
• Google Pixel 3a (Android 9)
• Huawei Mate 20 Lite (Android 8.1.0)
• Huawei P9 Lite (Android 7.0)
• OnePlus A3 (Android 6.0.1)
